test/framework/machinedeployment_helpers.go

```diff
@@ -306,7 +306,7 @@ func ScaleAndWaitMachineDeployment(ctx context.Context, input ScaleAndWaitMachin
 	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling ScaleAndWaitMachineDeployment")
 	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling ScaleAndWaitMachineDeployment")
 
-	log.Logf("Scaling machine deployment %s/%s from %v to %v replicas", input.MachineDeployment.Namespace, input.MachineDeployment.Name, input.MachineDeployment.Spec.Replicas, input.Replicas)
+	log.Logf("Scaling machine deployment %s/%s from %d to %d replicas", input.MachineDeployment.Namespace, input.MachineDeployment.Name, *input.MachineDeployment.Spec.Replicas, input.Replicas)
```

Comment:
I noticed that this log output was returning a memory reference, so fixing this while I'm here!

Reply:
What happens if the replicas are nil here? Is this a possibility?

Reply:
@mboersma @CecileRobertMichon should we protect against nil Replicas value here? (wouldn't that be a catastrophic error that needs handling somewhere earlier in the flow, if it's not already?)

Reply:
maybe we can add an Expect above to make sure it's not nil?